src/web/rustdoc.rs

```diff
@@ -666,4 +671,21 @@ mod test {
             Ok(())
         })
     }
+    #[test]
+    fn crate_name_percent_decoded_redirect() {
+        wrapper(|env| {
+            env.db()
+                .fake_release()
+                .name("fake-crate")
+                .version("0.0.1")
+                .rustdoc_file("fake_crate/index.html", b"some content")
+                .create()
+                .unwrap();
+
+            let web = env.frontend();
+            assert_redirect("/fake%2Dcrate", "/fake-crate/0.0.1/fake_crate/", web)?;
```

Comment:
Can you also test `/crate/fake%2Dcrate`?

Also, I think this will break for `/fake%2Dcrate/0.0.1/fake_crate`, but that can be added in a follow-up PR.